src/mlir/generator.cc

```diff
@@ -247,39 +255,19 @@ namespace mlir::verona
       return typeTable.insert(name, getMut(context));
     else if (name == "imm")
       return typeTable.insert(name, getImm(context));
-    else if (name == "bool")
-      return typeTable.insert(name, BoolType::get(context));
     else if (name == "unk")
       return typeTable.insert(name, UnknownType::get(context));
 
-    // Numeric (getAsInteger returns true on failure)
-    size_t bitWidth = 0;
-    if (name.substr(1).getAsInteger(10, bitWidth))
-      bitWidth = 64;
-    assert(bitWidth >= 8 && bitWidth <= 128 && "Bad numeric type width");
-
-    // Float (default literal `float` to F64)
-    // FIXME: This is not correct, but is needed to avoid opaque types
-    // We could leave unknown and resolve later as a pass
-    bool isFloat = name.startswith("F") || name == "float";
-    if (isFloat)
-      return typeTable.insert(name, FloatType::get(context, bitWidth));
-
-    // Integer (default literal `int` types to S64|U64)
-    // FIXME: This is not correct, but is needed to avoid opaque types
-    // We could leave unknown and resolve later as a pass
-    bool isSigned = name.startswith("S") || name == "int";
-    bool isUnsigned = name.startswith("U") || name == "hex" || name == "binary";
-    if (isSigned || isUnsigned)
-      return typeTable.insert(
-        name, IntegerType::get(context, bitWidth, isSigned));
-
-    // We need better error handling here, but this should never happen
-    llvm_unreachable("Cannot lower type");
+    // Every other type is just a class that we don't know yet
+    return typeTable.insert(name, ClassType::get(context, name));
   }
 
   mlir::Type Generator::parseType(const ::ast::Ast& ast)
   {
+    // Qualified types are references to classes with typeargs
```

Comment:
What are typeargs?

Reply:
Sorry, some comments refer to their AST nomenclature. I believe `typeargs` are the parameters, for example, generics. These are irrelevant in this context, I'll clean up the comment.